flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapMapState.java

```diff
@@ -140,25 +143,25 @@ public boolean contains(UK userKey) {
 	@Override
 	public Iterable<Map.Entry<UK, UV>> entries() {
 		Map<UK, UV> userMap = stateTable.get(currentNamespace);
-		return userMap == null ? null : userMap.entrySet();
+		return userMap == null ? emptyState.entrySet() : userMap.entrySet();
 	}
 
 	@Override
 	public Iterable<UK> keys() {
 		Map<UK, UV> userMap = stateTable.get(currentNamespace);
-		return userMap == null ? null : userMap.keySet();
+		return userMap == null ? emptyState.keySet() : userMap.keySet();
 	}
 
 	@Override
 	public Iterable<UV> values() {
 		Map<UK, UV> userMap = stateTable.get(currentNamespace);
-		return userMap == null ? null : userMap.values();
+		return userMap == null ? emptyState.values() : userMap.values();
```

Comment:
Why not using `Collections.emptyList()` ? 
The same for all above return results.